### PR TITLE
chore(query): add bloom filter supported data type

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Bloom filter can support more base data type:
Data
Timestamp

Closes #issue

### CODE REVIEW

File: src/query/storages/common/index/src/filters/xor8.rs [hunk 1]
```diff
@@ -128,9 +128,16 @@ impl Filter for Xor8Filter {
 
 impl SupportedType for Xor8Filter {
     fn is_supported_type(data_type: &DataType) -> bool {
-        // Bloom index only enabled for String and Integral types for now
         let inner_type = data_type.remove_nullable();
-        matches!(inner_type, DataType::Number(_) | DataType::String)
+        matches!(
+            inner_type,
+            DataType::Number(_)
+                | DataType::String
```
Comment (BohuTANG) [inline src/query/storages/common/index/src/filters/xor8.rs:135]: We used to support these types, but there were some performance issues(I can't find the PR ), so we removed them? cc @dantengsky

File: src/query/storages/common/index/src/filters/xor8.rs [hunk 2]
```diff
@@ -128,9 +128,16 @@ impl Filter for Xor8Filter {
...
+                | DataType::Timestamp
+                | DataType::Date
+                | DataType::Boolean
```
Comment (b41sh) [inline src/query/storages/common/index/src/filters/xor8.rs:138]: `Boolean` is better to use `MinMax` index

File: src/query/storages/common/index/src/filters/xor8.rs [hunk 3]
```diff
@@ -128,9 +128,16 @@ impl Filter for Xor8Filter {
...
+                | DataType::Variant
```
Comment (b41sh) [inline src/query/storages/common/index/src/filters/xor8.rs:139]: `Variant` usually filters on internal fields, so storing an index doesn't make much sense.
 The binary `JSONB` values are relatively large, and calculating the index may cause performance problems.

File: src/query/storages/common/index/src/filters/xor8.rs (hunk 1 above)
Comment (sundy-li) [reply src/query/storages/common/index/src/filters/xor8.rs]: why do we have performance issue about `timestamp,date` ?

Comment (BohuTANG) [reply src/query/storages/common/index/src/filters/xor8.rs]: Confirmed, it's my mistake. We only support `string` data type, please ignore my comments.

**Comment by BohuTANG** [comment]
@mergify update

File: src/query/storages/common/index/src/filters/xor8.rs (hunk 1 above)
Comment (dantengsky) [reply src/query/storages/common/index/src/filters/xor8.rs]: can not recall the details :(,  but it is not about performance (no benchmark of different data types has been done)

**Review by b41sh — APPROVED** [review]

**Pull request merged**